1. Research Question (研究問題)Before conducting the inferential analysis, we define the following statistical questions:
Proportion Analysis: "Is the proportion of high school students who felt sad or hopeless in 2007 different from the benchmark value of 0.30" 
Mean Analysis: "Is the mean BMI Percentile (BMIPCT) of students different from the benchmark value of 65.0" 

2. Variable Definition and Data Check (變數定義與資料檢查)In this section, we define the variables used and document the data cleaning process to ensure reproducibility:
2.1 Behavior Variable: SadOrHopelessDefinition: This variable measures whether students felt so sad or hopeless almost every day for two weeks or more in a row that they stopped doing some usual activities.Coding Rules:Success (1): The student felt sad or hopeless.Failure (0): The student did not feel sad or hopeless.
Data Cleaning: Missing or invalid values were handled using the .dropna() method to ensure analysis only includes valid responses.
2.2 Continuous Variable: BMIPCTDefinition: This variable represents the Body Mass Index (BMI) percentile of the student.Data Cleaning: We filtered the dataset to include only valid numerical entries for BMIPCT by removing missing values.2.3 Final Sample SizeAfter cleaning the data and removing missing values, the final sample sizes used for our analyses are:SadOrHopeless: n = 13,845$BMIPCT: n = 13,062


In [ ]:
import pandas as pd
from statsmodels.stats.proportion import proportions_ztest, proportion_confint

# 1. 準備數據
p0 = 0.30  # 根據老師要求的 Benchmark [cite: 40]
n_sad = len(df_sad)
count_sad = df_sad['is_sad'].sum()
p_hat = count_sad / n_sad

# 2. 建構信賴區間 
ci_low, ci_high = proportion_confint(count_sad, n_sad, alpha=0.05, method='normal')

# 3. 進行 Z 檢定 
z_stat, p_val = proportions_ztest(count_sad, n_sad, value=p0)

print(f"Sample Proportion (p-hat): {p_hat:.4f}")
print(f"95% Confidence Interval: ({ci_low:.4f}, {ci_high:.4f})")
print(f"Z-statistic: {z_stat:.4f}, P-value: {p_val:.4f}")

Sample Proportion (p-hat): 0.3000
95% Confidence Interval: (0.2923, 0.3076)
Z-statistic: -0.0093, P-value: 0.9926


Interpretation (結果解釋)主數值： 樣本比例為 30.00% 。CI 含義： 我們有 95% 的信心，全體學生感到悲傷的真實比例落在此區間內 。檢定結果： 由於 P\text{-value} > 0.05，我們 Fail to reject H_0。這代表目前沒有顯著證據支持該比例與 30% 有所不同 。


In [ ]:
from scipy import stats

# 1. 準備數據
mu0 = 65.0  # 根據老師要求的 BMIPCT Benchmark 
n_bmi = len(df_bmi)
mean_bmi = df_bmi['BMIPCT'].mean()
std_bmi = df_bmi['BMIPCT'].std() # 必須包含標準差 
sem_bmi = stats.sem(df_bmi['BMIPCT'])

# 2. 建構信賴區間 
ci_mu = stats.t.interval(0.95, df=n_bmi-1, loc=mean_bmi, scale=sem_bmi)

# 3. 進行單一樣本 T 檢定 
t_stat, p_val_t = stats.ttest_1samp(df_bmi['BMIPCT'], popmean=mu0)

print(f"Sample Mean: {mean_bmi:.4f}, Std Dev: {std_bmi:.4f}")
print(f"95% Confidence Interval: ({ci_mu[0]:.4f}, {ci_mu[1]:.4f})")
print(f"T-statistic: {t_stat:.4f}, P-value: {p_val_t:.4f}")

Sample Mean: 64.8207, Std Dev: 27.5168
95% Confidence Interval: (64.3487, 65.2926)
T-statistic: -0.7448, P-value: 0.4564


Final Synthesis
本研究旨在探討 2007 年高中生的心理健康比例與 BMI 分佈狀況。在比例分析中，我們發現感到悲傷或絕望的學生比例約為 30％，與預期基準一致。在平均數分析中，學生的平均 BMI 百分位數約為 64.82
以單一樣本 t 檢定證實其與基準值 65.0 無顯著差異。
綜合 EDA 的圖表觀察與推論統計結果，我們得出結論：該群體在心理感受比例與體重指標平均值上，皆符合設定的參考基準。這顯示該年度學生的相關健康指標處於預期範圍內 。

In [ ]:
# Create a summary table for the Final Report
summary_data = {
    "Variable": ["SadOrHopeless", "BMIPCT"],
    "Benchmark (H0)": [0.30, 65.0],
    "Sample Estimate": [0.3000, 64.8207],
    "95% Conf. Interval": ["(0.2923, 0.3076)", "(64.3487, 65.2926)"],
    "P-value": [0.9926, 0.4564],
    "Decision": ["Fail to Reject H0", "Fail to Reject H0"]
}
df_summary = pd.DataFrame(summary_data)
display(df_summary)


,Variable,Benchmark (H0),Sample Estimate,95% Conf. Interval,P-value,Decision
0,SadOrHopeless,0.3,0.3000,"(0.2923, 0.3076)",0.9926,Fail to Reject H0
1,BMIPCT,65.0,64.8207,"(64.3487, 65.2926)",0.4564,Fail to Reject H0
